ament_cmake_core/cmake/symlink_install/ament_cmake_symlink_install.cmake.in

```diff
@@ -240,7 +240,7 @@ function(ament_cmake_symlink_install_targets)
     get_filename_component(fileext "${file}" EXT)
     if(fileext STREQUAL ".a" OR fileext STREQUAL ".lib")
       set(destination "${ARG_ARCHIVE_DESTINATION}")
-    elseif(fileext STREQUAL ".dylib" OR fileext STREQUAL ".so")
+    elseif(fileext STREQUAL ".dylib" OR fileext MATCHES "\.so(\.[0-9]+)?(\.[0-9]+)?(\.[0-9]+)?$")
```

Comment:
Please still use `\d` for digits as suggested before.

Reply:
unfortunately, CMake regex doesn't support the `\d` syntax, I tested it this's what it gives me 

```
when parsing string

 \.so(\.\d+)?(\.\d+)?(\.\d+)?$

Invalid character escape '\d'.
```
And the [doc](https://cmake.org/cmake/help/v3.0/command/string.html) doesn't mention it